# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/7enno/HananAlawawdaRepository/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I chose the Refresh / Content Opportunity Scoring lane because I already worked with similar tasks in the starter notebooks, so I understand its main idea. It is easier for me to follow and apply, and I am interested in learning how observable signals can be used to prioritize pages for review.

In [25]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

My work aims to improve the decision of which pages should be reviewed first for content refresh. The recommendations can help the content or SEO team prioritize their work based on observable signals instead of reviewing pages randomly. A wrong recommendation could waste time by sending reviewers to pages that do not need attention or cause important pages to be overlooked.


In [26]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [27]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
!git clone https://github.com/7enno/HananAlawawdaRepository.git

import pandas as pd
import os

print(os.path.exists("data/raw/content_refresh_anonymized.csv"))

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")
print("Number of pages:", len(df))

print("Pages with declining trend:",
      (df["trend_direction"] == "down").sum())

print("Pages with at least 500 impressions:",
      (df["impressions_90d"] >= 500).sum())


Cloning into 'HananAlawawdaRepository'...
remote: Enumerating objects: 127, done.
remote: Counting objects: 100% (127/127), done.
remote: Compressing objects: 100% (84/84), done.
remote: Total 127 (delta 40), reused 92 (delta 27), pack-reused 0 (from 0)
Receiving objects: 100% (127/127), 1.85 MiB | 8.64 MiB/s, done.
Resolving deltas: 100% (40/40), done.
True
Number of pages: 30000
Pages with declining trend: 16262
Pages with at least 500 impressions: 16726


The starter dataset contains 30,000 pages. Among them, 16,262 pages have a declining trend, and 16,726 pages have at least 500 impressions. These numbers suggest that there are many pages that could benefit from prioritization, making the Refresh / Content Opportunity Scoring lane a meaningful problem to explore over the next seven weeks.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

My work will provide decision support by identifying pages that may be good candidates for content review based on observable signals. The results can show patterns and directional relationships that help prioritize pages for review. However, my work cannot prove that refreshing a page will cause better performance, predict Google's algorithms, or establish causal relationships. Human review is still needed before taking action.

My work will provide decision support by identifying pages that may be good candidates for content review based on observable signals. The results can show patterns and directional relationships that help prioritize pages for review. However, my work cannot prove that refreshing a page will cause better performance, predict Google's algorithms, or establish causal relationships. Human review is still needed before taking action.

In [28]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.